### ISSUE TITLE
[libc++][format] String literals inside of `tuple` and `pair` should be escaped

### ISSUE THREAD

**Issue by JMazurkiewicz** [llvm/llvm-project#59763]
Example:
```c++
#include <cstdio>
#include <format>
#include <tuple>
#include <string>
#include <string_view>
#include <utility>

using namespace std;

int main() {
    const tuple t{"_\t_", "_\t_"s};
    const pair p{auto{"_\t_"}, "_\t_"sv};
    puts(format("{}\n{}", t, p).c_str());
}
```

Expected output:
```
("_\t_", "_\t_")
("_\t_", "_\t_")
```

Got:
```
(_	_, "_\t_")
(_	_, "_\t_")
```

Compiler explorer: https://godbolt.org/z/8MnaMzzvo
Debug-enabled specializations: [[format.formatter.spec]/2.2](http://eel.is/c++draft/format.formatter#spec-2.2)
Parsing format spec for `tuple` and `pair`: [[format.tuple]/7](http://eel.is/c++draft/format.tuple#7)

**Comment by mordante** [comment]
Thanks for the report! I noticed this before and have a fix in https://reviews.llvm.org/D140653#change-EQC0Ek1EVC2k
I now created https://reviews.llvm.org/D140819 which fixes the issue and adds regression tests.

It was only for string literals, this uses a different formatter, which doesn't determine the length of the string when it's not needed. However I had forgotten to add the escape formatting test to this formatter.
